pyvisa_py/tcpip.py

```diff
@@ -1273,6 +1273,37 @@ def clear(self) -> StatusCode:
 
         return StatusCode.success
 
+    def flush(self, mask: BufferOperation) -> StatusCode:
+        """Flush the specified buffers.
+        Corresponds to viFlush function of the VISA library.
+        Parameters
+        ----------
+        mask : constants.BufferOperation
+            Specifies the action to be taken with flushing the buffer.
+            The values can be combined using the | operator. However multiple
+            operations on a single buffer cannot be combined.
+        Returns
+        -------
+        constants.StatusCode
+            Return value of the library call.
+        """
+        if (
+            mask & BufferOperation.discard_read_buffer
+            or mask & BufferOperation.discard_read_buffer_no_io
+            or mask & BufferOperation.discard_receive_buffer
+            or mask & BufferOperation.discard_receive_buffer2
+        ):
+            return self.clear()
+        if (
+            mask & BufferOperation.flush_write_buffer
+            or mask & BufferOperation.flush_transmit_buffer
+            or mask & BufferOperation.discard_write_buffer
+            or mask & BufferOperation.discard_transmit_buffer
+        ):
+            pass
```

Comment:
I likely do not understand enough of sockets to conclude if one here should wait for the socket to get writable via a call to select:

```
_, w, _ = select.select(,[], [self.interface], [], timeout)
if not w:
    return some_error_code
```

I see further no way to distinguish flush from discard operations

Reply:
The way write is implemented we always ensure all data are sent so there is never anything to flush I would say. I think your implementation is fine as long as we do not explicitly add buffered write.